### PR TITLE
Update Azure NetApp management and add NFS protocol version

### PR DESCRIPTION
**Pull request by wmudge**
Signed-off-by: Webster Mudge <wmudge@cloudera.com>

### CODE REVIEW

**Review by jimright — COMMENTED** [review]
Did a simple test of this PR to confirm CDP Azure deployment works correctly.
Steps for test:
* CDP Azure definition included a CML workspace with default root volume size
* Confirmed that Azure NetApp Volume was created with NFSv3 protocol type. Expected from updates.
* Confirmed that {{run__azure_netapp_nfs_mount}} was set and used during the "Construct CDP ML Workspace configurations" task in runtime.
* ML workspace was successfully deployed and Filesystem ID listed in Workspace details matches the NetApp volume mount path.

File: roles/runtime/tasks/initialize_base.yml [hunk 1]
```diff
@@ -152,7 +152,8 @@
         base_instance_group: "{{ run__ml_k8s_request_base }}"
         config:
           name: "{{ __ml_config.name | default([run__namespace_cdp, __ml_config.suffix | default(include.suffix) | default(run__ml_suffix)] | join('-')) }}"
-          nfs: "{{ __ml_config.nfs | default(run__azure_nfs_mount | default(omit)) }}"
+          nfs: "{{ __ml_config.nfs | default(run__azure_netapp_nfs_mount | default(omit)) }}"
+          nfs_version: "{{ __ml_config.nfs_version | default(run__azure_netapp_nfs_version if __ml_config.nfs is defined or run__azure_netapp_nfs_mount is defined | default(omit)) }}"
```
Comment (jimright) [inline roles/runtime/tasks/initialize_base.yml:156]: When testing `teardown` I get an error on this task that is related to the if expression on this line:

**Error message (snippet):**

```
TASK [cloudera.exe.runtime : Construct CDP ML Workspace configurations] ********
task path: /runner/project/ansible_dev/collections/ansible_collections/cloudera/exe/roles/runtime/tasks/initialize_base.yml:144
Wednesday 30 March 2022  14:02:34 +0000 (0:00:00.170)       0:01:02.875 *******
fatal: [localhost]: FAILED! => {
    "msg": "The task includes an option with an undefined variable. The error was: <snipped>....: 
the inline if-expression on line 1 evaluated to false and no else section was defined.
}
```

**CML definition:**

```
ml:
  definitions:
    - name: "{{ name_prefix }}-default-ml"
      tls: yes
      monitoring: yes
      governance: yes
      metrics: yes
      public_loadbalancer: yes
      storage: yes
      instance_groups:
        - name: cml
          instanceCount: 0
          instanceType: "Standard_DS5_v2"
          rootVolume:
            size: 100
```

**Review by jimright — CHANGES_REQUESTED** [review]
@wmudge,  

Hit an error during the environment teardown. Seems to be related to the changes made in `roles/runtime/tasks/initialize_base.yml`. See inline comment for details.

**Comment by jimright** [comment]
Thanks @wmudge - commit [86b519f](https://github.com/cloudera-labs/cloudera.exe/pull/86/commits/86b519fc6c85acb533e08dbce785ca99f18cca18) resolved the teardown error in platform.

I hit another error during infrastructure teardown where the NetApp Capacity Pool could not be removed due to nested resources still being present. When I re-ran teardown this completed successfully so I suspect this is either an eventual consistency issue or a wait is needed on a previous task (e.g. [Remove Azure NetApp Volume in cloudera.exe.infrastructure teardown_azure_storage.yml](https://github.com/cloudera-labs/cloudera.exe/blob/devel/roles/infrastructure/tasks/teardown_azure_storage.yml#L33) 

Snippet of error message is below.  

```
TASK [cloudera.exe.infrastructure : Remove NetApp Capacity Pool] ***************
task path: /runner/project/ansible_dev/collections/ansible_collections/cloudera/exe/roles/infrastructure/tasks/teardown_azure_storage.yml:41
Wednesday 30 March 2022  16:34:03 +0000 (0:02:08.831)       0:36:25.888 *******
<snipped>
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_netapp.azure.azure_rm_netapp_capacity_pool_payload_nfvqr2vg/ansible_netapp.azure.azure_rm_netapp_capacity_pool_payload.zip/ansible_collections/netapp/azure/plugins/modules/azure_rm_netapp_capacity_pool.py", line 204, in delete_azure_netapp_capacity_pool
  File "/usr/local/lib/python3.8/site-packages/azure/mgmt/netapp/operations/_pools_operations.py", line 549, in begin_delete
    raw_result = self._delete_initial(
  File "/usr/local/lib/python3.8/site-packages/azure/mgmt/netapp/operations/_pools_operations.py", line 505, in _delete_initial
    map_error(status_code=response.status_code, response=response, error_map=error_map)
  File "/usr/local/lib/python3.8/site-packages/azure/core/exceptions.py", line 105, in map_error
    raise error
azure.core.exceptions.ResourceExistsError: (CannotDeleteResource) Can not delete resource before nested resources are deleted.
Code: CannotDeleteResource
Message: Can not delete resource before nested resources are deleted.
```

**Comment by wmudge** [comment]
I suspect this is an eventual consistency issue and will need to add a guard condition. Still looking into it!

**Review by jimright — APPROVED** [review]
@wmudge - confirmed that both setup and teardown of Azure environment with CML workspace now works with your latest changes. Looks good.

**Pull request merged**